test/test_hpytype.py

```diff
@@ -267,6 +267,7 @@ def test_HPyDef_Member_integers(self):
                 ('HPYSSIZET', 'HPy_ssize_t'),
                 ]:
             mod = self.make_module("""
+            #include <Python.h>
```

Comment:
I really would like to keep all the tests, because otherwise its too easy to only check that features work with pure type and too much extra work to add tests for the different cases.

If it's an issue for PyPy, perhaps what I can do is move all the legacy tests into separate files. So rename `test_cpy_compat` to `test_legacy` and then add `test_hpytype_legacy` and `test_slot_legacy`? Or even put all the legacy tests into `test_legacy`?

Reply:
I think that putting everything in `test_legacy.py` should be enough.
Probably we also want to introduce a special class `HPyLegacyTest` (similar to `HPyDebugTest`) which for now it's just a subclass of `HPyTest` and nothing else.
The difference between the two is that once we have an official "hybrid" ABI, `HPyTest` tests will run with `[cpython]` and `[universal]`, while `HPyLegacyTest`'s will run with `[cpython]` and `[hybrid]`

Reply:
Sounds good. I will reshuffle the tests.

Reply:
thanks!

Reply:
Refactoring done. The three test files that use `Python.h` are now `test_cpy_compat.py`, `test_hpytype_legacy.py` and `test_slots_legacy.py`. I also moved the legacy type and slot tests from `test_cpy_compat.py` into the new files.